pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -623,7 +623,7 @@ def _handle_horovod(self) -> None:
         hvd.init()
         if isinstance(self.accelerator, GPUAccelerator):
             # Horovod assigns one local GPU per process
-            self._parallel_devices = list(range(hvd.local_size()))
+            self._parallel_devices = [torch.device(f"cuda:{i}") for i in range(hvd.local_size())]
```

Comment:
Does hvd expect a list of id or a list of cuda devices? Both works right?
+1 to this change if both work, cuda devices list is more consistent with other strategies' parallel devices

Reply:
This is required to be device since the value will propegate to root_device used here: https://github.com/PyTorchLightning/pytorch-lightning/blob/3079416ae5aaecad80511d901454096a00ce2c88/pytorch_lightning/strategies/horovod.py#L140